compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerToUKernels.cpp

```diff
@@ -159,12 +159,36 @@ static FailureOr<IREE::Codegen::UKernelOpInterface> matchDAGForUKernel(
   }
 
   Location loc = op.getLoc();
+  Type i64 = rewriter.getI64Type();
+
+  // The ukernel requires a padding value of type i64. When the element type is
+  // a narrower N-bit type, only the least significant N bits of the i64 padding
+  // value are used.
   Value paddingVal = op.getPaddingValue();
+  // If the pack op didn't have a padding_value attribute, default to 0.
   if (!paddingVal) {
-    paddingVal = rewriter.create<arith::ConstantOp>(
-        loc, rewriter.getZeroAttr(inElemType), inElemType);
+    paddingVal =
+        rewriter.create<arith::ConstantOp>(loc, rewriter.getZeroAttr(i64), i64);
+  }
+  // Element types > 64bits could be supported, when the padding value is a
+  // repeating 64-bit pattern. For now, we leave this as not-yet-implemented.
+  int paddingValBitWidth = paddingVal.getType().getIntOrFloatBitWidth();
+  if (paddingValBitWidth > 64) {
+    return rewriter.notifyMatchFailure(op,
+                                       "unsupported padding_value bit width");
+  }
+  // Non-integer element types get bitcast to integer of same bit width.
+  if (!paddingVal.getType().isSignlessInteger()) {
+    Type sameWidthIntType = rewriter.getIntegerType(paddingValBitWidth);
+    paddingVal =
+        rewriter.create<arith::BitcastOp>(loc, sameWidthIntType, paddingVal);
+  }
+  // Integers narrower than 64 bit get extended to 64 bits, it doesn't matter
+  // how, as the high bits are unused.
+  if (paddingValBitWidth < 64) {
+    paddingVal =
+        rewriter.create<arith::ExtUIOp>(loc, rewriter.getI64Type(), paddingVal);
```

Comment:
We can use `i64` variable here. Semantically, we should use the variable as same as the default padding value creation, i.e., line 171.